packages/noya-colorpicker/src/hooks/useEventCallback.ts

```diff
@@ -1,14 +1,20 @@
-import { useRef, useEffect, useCallback } from "react";
+import { useRef, useEffect, useCallback } from 'react';
 
 // Saves incoming handler to the ref in order to avoid "useCallback hell"
-function useEventCallback<T>(handler?: (value: T) => void): (value: T) => void {
+function useEventCallback<T>(
```

Comment:
I think we can use generics to handle both cases:

```ts
function useEventCallback<T extends any[]>(handler?: (...values: T) => void): (...values: T) => void {
  const callbackRef = useRef(handler);

  useEffect(() => {
    callbackRef.current = handler;
  });

  return useCallback((...values: T) => callbackRef.current && callbackRef.current(...values), []);
}
```

If we remove the type annotations from `useEventCallback<string>(...)` => `useEventCallback(...)` I think they should be inferred correctly. We can do `useEventCallback<[string]>(...)` if we still need type annotations in some cases